### PR TITLE
Add virtual keyword for every get child resource method

### LINKED ISSUE #1718: All `public` methods should be `virtual`
Currently we are missing `virtual` in the method that returns a `collection` or a singleton `resource`

### PR DESCRIPTION
**Pull request by Yao725**
Fix #1718 

# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs [hunk 1]
```diff
@@ -472,7 +472,7 @@ protected override void WriteResourceCollectionEntry(Resource resource)
             _writer.WriteXmlDocumentationReturns($"An object representing collection of {resource.Type.Name.LastWordToPlural()} and their operations over a {_resource.Type.Name}.");
             _writer.WriteXmlDocumentationParameters(collection.ExtraConstructorParameters);
             var extraConstructorParameters = collection.ExtraConstructorParameters;
-            _writer.Append($"public {collection.Type.Name} Get{resource.Type.Name.ResourceNameToPlural()}(");
+            _writer.Append($"public {GetVirtual(true)} {collection.Type.Name} Get{resource.Type.Name.ResourceNameToPlural()}(");
```
Comment (AlexanderSher) [inline src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs:475]: if `GetVirtual` is always called with `true`, why not just add `virtual` to the string directly?

Comment (AlexanderSher) [reply src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs:475]: And while you're here, please change `{collection.Type.Name}` to `{collection.Type}`.

Comment (Yao725) [reply src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs:475]: > if `GetVirtual` is always called with `true`, why not just add `virtual` to the string directly?

Use `GetVirtual` to keep consistency, like [here](https://github.com/Azure/autorest.csharp/blob/9d6a700eea66d61eb5740af40ff7a1cc5e63f1ea/src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs#L317)

Comment (Yao725) [reply src/AutoRest.CSharp/Mgmt/Generation/ResourceWriter.cs]: > And while you're here, please change `{collection.Type.Name}` to `{collection.Type}`.

Changed

**Review by fengzhou-msft — APPROVED** [review]

**Review by archerzz — APPROVED** [review]

**Pull request merged**